src/main/scala/ai/privado/utility/Utilities.scala

```diff
@@ -187,11 +188,18 @@ object Utilities {
           .slice(startLine - 1, endLine)
           .zipWithIndex
           .map { case (line, lineNo) =>
-            if (lineToHighlight.isDefined && lineNo == lineToHighlight.get - startLine) {
-              line + " " + arrow
+            // Zero means no truncate required
+            val modifiedLine = if (allowedLineLimit != 0 && line.length > allowedLineLimit) {
+              line.take(allowedLineLimit) + "..."
             } else {
               line
             }
+
+            if (lineToHighlight.isDefined && lineNo == lineToHighlight.get - startLine) {
+              modifiedLine + " " + arrow
+            } else {
+              modifiedLine
+            }
```

Comment:
create a utility function 
```
def getTruncatedText(text: String, allowedCharLimit: Option[Int] = None) : String
```
and use it where ever we need to truncate the text

Reply:
done.